x-pack/plugins/observability/public/pages/slos/components/toggle_slo_view.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButtonGroup,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverTitle,
+} from '@elastic/eui';
+import { GridSize } from './slos_grid/grid_size';
+
+export type SLOViewType = 'cardView' | 'listView';
+
+interface Props {
+  setGridSize: (gridSize?: string) => void;
+  setSLOView: (view: SLOViewType) => void;
+  sloView: SLOViewType;
+}
+const toggleButtonsIcons = [
+  {
+    id: `cardView`,
+    label: 'Card View',
+    iconType: 'visGauge',
+    'data-test-subj': 'sloCardViewButton',
+  },
+  {
+    id: `listView`,
+    label: 'List View',
+    iconType: 'list',
+    'data-test-subj': 'sloListViewButton',
+  },
+];
+
+export function ToggleSLOView({ sloView, setSLOView, setGridSize }: Props) {
+  return (
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiButtonGroup
+          legend="Text align"
+          options={toggleButtonsIcons}
+          idSelected={sloView}
+          onChange={(id) => setSLOView(id as SLOViewType)}
+          isIconOnly
+        />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <ViewSettings setGridSize={setGridSize} />
```

Comment:
How about we show the settings icon only on card view? Currently it appears on the list view as well

<img width="1716" alt="Screenshot 2023-11-21 at 09 50 37" src="https://github.com/elastic/kibana/assets/2852703/26427b20-b9f1-4d73-8467-8d029a75074f">